### ISSUE TITLE
Creating new tab in Safari makes iPhone reboot

### ISSUE THREAD

**Issue by Jonathan0827** [roothide/Bootstrap#237]
### Device Model

iPhone 13, A15

### iOS Version

17.0

### Bootstrap Version

1.1

### What happened?

Enabled tweak injection of Safari. Creating new tab in Safari made iPhone reboot

### Acknowledgement of README and FAQ

- [X] I have read both the README and the FAQ.

**Comment by dleovl** [comment]
Confirmed, same thing here, same iOS and Bootstrap ver. Weird!

**Comment by dleovl** [comment]
For the record, this also happens when not bootstrapped (though the app is still injected; it can launch). Probably just due to the method of tweak injection.

**Comment by dleovl** [comment]
Hm, seems like this issue fixes itself after clearing Safari data. Can you confirm it isn't from a lot of tabs / data? @Jonathan0827 Looking good on my end!

**Comment by dleovl** [comment]
This issue seems to be reproducable by
1. Clearing Safari data
2. Opening Safari (create some activity maybe?)
3. Force close Safari from App Switcher
4. Open Safari and make a new tab
It seems when the device reboots into stock it doesn't happen again. cc @p2kdev

**Comment by dleovl** [comment]
.. just had Safari reboot the phone while being unbootstrapped and opening Safari back up from it being in the background! I think it's safe to say you probably shouldn't have Safari injected at all if any of these are happening to you.

**Comment by dleovl** [comment]
Reboot does not produce any logs, none viewable in KrashKop and none in CrashReporter.

**Comment by roothider** [comment]
this issue cannot be reproduced in iphone13pm 17.0 ?

**Comment by Johnsmith927** [comment]
I also have this issue on iPhone 13 iOS 17.0. When bootstrapped, when I press enter on a search, my iPhone reboots unbootstrapped. Toggling off safari in the app list fixes this.

**Comment by roothider** [comment]
> I also have this issue on iPhone 13 iOS 17.0. When bootstrapped, when I press enter on a search, my iPhone reboots unbootstrapped. Toggling off safari in the app list fixes this.

this issue cannot be reproduced in my iphone13pm 17.0, maybe it's caused by some tweak, you can try this to see if Safari still causes the device to reboot:
1: enable tweak for safari in the AppList of Bootstrap
2: disable tweak injection in the Settings of Bootstrap
3: launch safari app

**Comment by Johnsmith927** [comment]
Yeah it still shuts down with tweaks disabled in bootstrap. For me it happens when I switch tab groups and then try to search something pretty shortly afterwards, and I think yesterday it happened because I launched safari and then quickly after made a search. Do you think I should just wait a couple of seconds after some activity in safari before doing a search?

**Comment by roothider** [comment]
> Yeah it still shuts down with tweaks disabled in bootstrap. For me it happens when I switch tab groups and then try to search something pretty shortly afterwards, and I think yesterday it happened because I launched safari and then quickly after made a search. Do you think I should just wait a couple of seconds after some activity in safari before doing a search?
> 
> Edit: I got it to shut down twice just now, once with bootstrap tweaks and safari injection enabled and once with bootstrap no tweaks and no safari injection. Edit 2: meant the second time safari injection was on, but tweak injection was off in bootstrap settings. Sorry for the confusion

If this is the case, then I suggest you not to enable tweaks for Safari, as iOS 17.0 itself has some wierd BUGs

**Comment by Symplicityy** [comment]
This is also happening to me even with all tweaks disabled through choicy. I’m using an iPhone 15 Pro Max 17.0

**Comment by IvanIVGrozny** [comment]
Happens with iPhone 15 Pro, 17.0 21A327. This is disastrous! I need my custom WebKit fonts 😭

**Comment by roothider** [comment]
> Happens with iPhone 15 Pro, 17.0 21A327. This is disastrous! I need my custom WebKit fonts 😭

stock bug of ios17

**Comment by roothider** [comment]
are there any panic logs in rootfs:/var/mobile/Library/Logs/CrashReporter/ ?

**Comment by Symplicityy** [comment]
> are there any panic logs in rootfs:/var/mobile/Library/Logs/CrashReporter/ ?

Yes, there's a kernel panic related to safari. [](https://cdn.discordapp.com/attachments/424451479638769675/1289618275323478170/panic-full-2024-09-28-115345.ips?ex=66f97a1c&is=66f8289c&hm=e1452936c38b135fffd6cef6fbf92ad53f3ce48c1e1edd591444fd23088e4154&)

**Comment by roothider** [comment]
> > are there any panic logs in rootfs:/var/mobile/Library/Logs/CrashReporter/ ?
> 
> Yes, there's a kernel panic related to safari. https://cdn.discordapp.com/attachments/424451479638769675/1289618275323478170/panic-full-2024-09-28-115345.ips?ex=66f97a1c&is=66f8289c&hm=e1452936c38b135fffd6cef6fbf92ad53f3ce48c1e1edd591444fd23088e4154&

seems to be a new security mechanism for iOS 17.0 on certain devices, can you upload the rootfs:/sbin/launchd file of your device ?

**Comment by Symplicityy** [comment]
> > > are there any panic logs in rootfs:/var/mobile/Library/Logs/CrashReporter/ ?
> 
> > 
> 
> > Yes, there's a kernel panic related to safari. https://cdn.discordapp.com/attachments/424451479638769675/1289618275323478170/panic-full-2024-09-28-115345.ips?ex=66f97a1c&is=66f8289c&hm=e1452936c38b135fffd6cef6fbf92ad53f3ce48c1e1edd591444fd23088e4154&
> 
> 
> 
> seems to be a new security mechanism for iOS 17.0 on certain devices, can you upload the rootfs:/sbin/launchd file of your device ?

[https://cdn.discordapp.com/attachments/424451479638769675/1289625635995516980/launchd?ex=66f980f7&is=66f82f77&hm=3aaf313cd1d86f3855bd708b56824e25372aaaba1250fbc03f6c28a1d04ed3ab&](https://cdn.discordapp.com/attachments/424451479638769675/1289625635995516980/launchd?ex=66f980f7&is=66f82f77&hm=3aaf313cd1d86f3855bd708b56824e25372aaaba1250fbc03f6c28a1d04ed3ab&)

**Comment by roothider** [comment]
> > > > are there any panic logs in rootfs:/var/mobile/Library/Logs/CrashReporter/ ?
> > 
> > 
> > > 
> > 
> > 
> > > Yes, there's a kernel panic related to safari. https://cdn.discordapp.com/attachments/424451479638769675/1289618275323478170/panic-full-2024-09-28-115345.ips?ex=66f97a1c&is=66f8289c&hm=e1452936c38b135fffd6cef6fbf92ad53f3ce48c1e1edd591444fd23088e4154&
> > 
> > 
> > seems to be a new security mechanism for iOS 17.0 on certain devices, can you upload the rootfs:/sbin/launchd file of your device ?
> 
> https://cdn.discordapp.com/attachments/424451479638769675/1289625635995516980/launchd?ex=66f980f7&is=66f82f77&hm=3aaf313cd1d86f3855bd708b56824e25372aaaba1250fbc03f6c28a1d04ed3ab&

try this beta:  https://cdn.discordapp.com/attachments/1002450165443072041/1289670430654988400/Bootstrap.tipa?ex=66f9aaaf&is=66f8592f&hm=9069003f1bc634476d63f60bc462c76ff63290208e49974c3b7185ff8ee3e62e&

1: reboot device
2: install beta version and bootstrap your device
3: disable tweak for safari
4: re-enable tweak for safari

**Comment by Symplicityy** [comment]
> > > > > are there any panic logs in rootfs:/var/mobile/Library/Logs/CrashReporter/ ?
> 
> > > 
> 
> > > 
> 
> > > > 
> 
> > > 
> 
> > > 
> 
> > > > Yes, there's a kernel panic related to safari. https://cdn.discordapp.com/attachments/424451479638769675/1289618275323478170/panic-full-2024-09-28-115345.ips?ex=66f97a1c&is=66f8289c&hm=e1452936c38b135fffd6cef6fbf92ad53f3ce48c1e1edd591444fd23088e4154&
> 
> > > 
> 
> > > 
> 
> > > seems to be a new security mechanism for iOS 17.0 on certain devices, can you upload the rootfs:/sbin/launchd file of your device ?
> 
> > 
> 
> > https://cdn.discordapp.com/attachments/424451479638769675/1289625635995516980/launchd?ex=66f980f7&is=66f82f77&hm=3aaf313cd1d86f3855bd708b56824e25372aaaba1250fbc03f6c28a1d04ed3ab&
> 
> 
> 
> try this beta:  https://cdn.discordapp.com/attachments/1002450165443072041/1289670430654988400/Bootstrap.tipa?ex=66f9aaaf&is=66f8592f&hm=9069003f1bc634476d63f60bc462c76ff63290208e49974c3b7185ff8ee3e62e&
> 
> 
> 
> 1: reboot device
> 
> 2: install beta version and bootstrap your device
> 
> 3: disable tweak for safari
> 
> 4: re-enable tweak for safari
> 
> 
> 
> 

It works now! Safari doesn't cause a kernel panic anymore when creating a new tab. The only thing that still isn't working properly is single tab mode. For some reason it doesn't care what you have it set to in settings, it always uses the tab bar.

**Comment by roothider** [comment]
> > > > > > are there any panic logs in rootfs:/var/mobile/Library/Logs/CrashReporter/ ?
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > >
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > > Yes, there's a kernel panic related to safari. https://cdn.discordapp.com/attachments/424451479638769675/1289618275323478170/panic-full-2024-09-28-115345.ips?ex=66f97a1c&is=66f8289c&hm=e1452936c38b135fffd6cef6fbf92ad53f3ce48c1e1edd591444fd23088e4154&
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > seems to be a new security mechanism for iOS 17.0 on certain devices, can you upload the rootfs:/sbin/launchd file of your device ?
> > 
> > 
> > > 
> > 
> > 
> > > https://cdn.discordapp.com/attachments/424451479638769675/1289625635995516980/launchd?ex=66f980f7&is=66f82f77&hm=3aaf313cd1d86f3855bd708b56824e25372aaaba1250fbc03f6c28a1d04ed3ab&
> > 
> > 
> > try this beta:  https://cdn.discordapp.com/attachments/1002450165443072041/1289670430654988400/Bootstrap.tipa?ex=66f9aaaf&is=66f8592f&hm=9069003f1bc634476d63f60bc462c76ff63290208e49974c3b7185ff8ee3e62e&
> > 1: reboot device
> > 2: install beta version and bootstrap your device
> > 3: disable tweak for safari
> > 4: re-enable tweak for safari
> 
> It works now! Safari doesn't cause a kernel panic anymore when creating a new tab. The only thing that still isn't working properly is single tab mode. For some reason it doesn't care what you have it set to in settings, it always uses the tab bar.

try this beta, should be fixed. (require relaunching safari):
https://cdn.discordapp.com/attachments/1002450165443072041/1289720697005805568/Bootstrap.tipa?ex=66f9d980&is=66f88800&hm=81dac96d48925acec1dde491df23025373aa3e3a4640f2bc1b6dfa8919b38ae9&

**Comment by Symplicityy** [comment]
> > > > > > > are there any panic logs in rootfs:/var/mobile/Library/Logs/CrashReporter/ ?
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > > >
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > > > Yes, there's a kernel panic related to safari. https://cdn.discordapp.com/attachments/424451479638769675/1289618275323478170/panic-full-2024-09-28-115345.ips?ex=66f97a1c&is=66f8289c&hm=e1452936c38b135fffd6cef6fbf92ad53f3ce48c1e1edd591444fd23088e4154&
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > > seems to be a new security mechanism for iOS 17.0 on certain devices, can you upload the rootfs:/sbin/launchd file of your device ?
> 
> > > 
> 
> > > 
> 
> > > > 
> 
> > > 
> 
> > > 
> 
> > > > https://cdn.discordapp.com/attachments/424451479638769675/1289625635995516980/launchd?ex=66f980f7&is=66f82f77&hm=3aaf313cd1d86f3855bd708b56824e25372aaaba1250fbc03f6c28a1d04ed3ab&
> 
> > > 
> 
> > > 
> 
> > > try this beta:  https://cdn.discordapp.com/attachments/1002450165443072041/1289670430654988400/Bootstrap.tipa?ex=66f9aaaf&is=66f8592f&hm=9069003f1bc634476d63f60bc462c76ff63290208e49974c3b7185ff8ee3e62e&
> 
> > > 1: reboot device
> 
> > > 2: install beta version and bootstrap your device
> 
> > > 3: disable tweak for safari
> 
> > > 4: re-enable tweak for safari
> 
> > 
> 
> > It works now! Safari doesn't cause a kernel panic anymore when creating a new tab. The only thing that still isn't working properly is single tab mode. For some reason it doesn't care what you have it set to in settings, it always uses the tab bar.
> 
> 
> 
> try this beta, should be fixed. (require relaunching safari):
> 
> https://cdn.discordapp.com/attachments/1002450165443072041/1289720697005805568/Bootstrap.tipa?ex=66f9d980&is=66f88800&hm=81dac96d48925acec1dde491df23025373aa3e3a4640f2bc1b6dfa8919b38ae9&

It worked!
